engines/supernova/screen.cpp

```diff
@@ -286,12 +287,16 @@ void Screen::renderMessage(int stringId, MessagePosition position,
 			text = Common::String::format(text.c_str(), var1.c_str());
 	}
 
+
 	renderMessage(text, position);
 }
 
 void Screen::renderMessage(const Common::String &text, MessagePosition position) {
 	if (!text.empty())
 		renderMessage(text.c_str(), position);
+	Common::TextToSpeechManager *ttsMan = g_system->getTextToSpeechManager();
+	if (ttsMan != nullptr && ConfMan.getBool("tts_enabled"))
+		ttsMan->say(text,  Common::TextToSpeechManager::INTERRUPT_NO_REPEAT);
 }
```

Comment:
Why did you add the TTS here and not in `Screen::renderMessage(const char *text, MessagePosition position, int positionX, int positionY)`?

That other render message is called from here, but it is also sometimes called directly. And when called directly it will be missing the TTS. So it would be better to move the TTS to that other one.

I am also wondering if queuing might not be better than interrupt here.

Reply:
Here is a savegame for Mission Supernova 2 that will show the issue. You will need to unzip it. After loading the savegame, go to the Entrance (top right) and then click through the dialog. You will see that you are missing the reaction of the audience, but if you move the TTS to the place I suggest it should work.
[ms2_save.014.zip](https://github.com/scummvm/scummvm/files/6979064/ms2_save.014.zip)